src/snowflake/snowpark/table.py

```diff
@@ -274,6 +274,12 @@ def __init__(
     def __copy__(self) -> "Table":
         return Table(self.table_name, self._session)
 
+    def __enter__(self):
```

Comment:
since we add them, let's have an example in docstring of `Table` class?

Reply:
Added text and samples in  the `cache_result` docstring because it's the main use case.